Automattic/woocommerce-payments
# feat: error message on 1M+ amount

Fixes https://github.com/Automattic/woocommerce-payments/issues/7485

#### Changes proposed in this Pull Request









Adds an error notice when the amount exceeds 1M in the settlement currency (EUR, USD, etc.).

#### Testing instructions





- Create a product to the cart to exceed the 1M (e.g.: $1000000 USD or 1000000€)
- Go to the blocks or shortcode checkout
- Select WooPayments as the processor
- The error should appear in place of the payment element component
  - You might notice there is a console error related to Stripe fingerprinting, but I wasn't able to resolve it 😬 
  - I wasn't able to remove the "save payment method" checkbox on blocks checkout

Blocks checkout with SEPA/Card
<img width="1318" alt="Screenshot 2024-05-07 at 6 43 06 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/5360dc68-091b-4778-90e3-6ac84c1a26a7">
<img width="1327" alt="Screenshot 2024-05-07 at 6 42 56 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/0177f9d7-8bdd-4514-a281-a2c8ff28e217">

Shortcode checkout with TT4 theme & SEPA
<img width="1308" alt="Screenshot 2024-05-07 at 6 11 29 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/120af8c6-9f3f-459e-9316-094c695dade3">

Shortcode checkout with Storefront theme & SEPA/Card
<img width="540" alt="Screenshot 2024-05-07 at 6 10 58 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/4b928858-56c7-42b2-8602-781c3bf8d434">
<img width="536" alt="Screenshot 2024-05-07 at 6 10 52 PM" src="https://github.com/Automattic/woocommerce-payments/assets/273592/45b0287c-1ce2-4eba-9958-7f4942d482b4">


-------------------


**Post merge**
